README.md
# DSMCalibrationData

Calibration data for CVPIA's Structure Decision Making chinook life cycle models.

This package contains the following calibration datasets:

* `calibration_year_index` and `calibration_year_spawn_index` the set of proxy years used to represent model inputs for 1997-2017 selected from the 1979-2000 simulation inputs.
* `grandtab_imputed` used as seeding values for calibration.
* `grandtab_observed` used for comparison to predicted values in optimization process.

and one function: 

* `set_synth_years` generates model inputs for running the life cycle models in calibration mode. For more information run `?set_synth_years` in the console. 

See the `data-raw/` directory to see the process for creating the datasets listed above.

## Dependencies
The `DSMCalibrationData` package uses data from several other packages within the [CVPIA Open Science Collaborative](https://github.com/CVPIA-OSC). These relationships are visualized in the dependency graph below. 
<img src="man/figures/dependencyChain.svg" width="100%"/>

<div style="margin-top: 40px;">Data Assembled and Maintained by <a href = "http://www.flowwest.com/" target = "_blank"> <img src="man/figures/TransLogoTreb.png" width="150px"/></div>


NEWS.md
# DSMCalibrationData 2.0

The 2022 update to DSMCalibrationData includes the following changes:

* Updates Yuba River escapement values to use passage data from a Vaki River Watch 
monitoring program instead of GrandTab data. For years that Vaki data is out we use original grand tab data. 


# DSMCalibrationData 1.0

GrandTab Calibration datasets for use with the following DSM models:

* [Fall Run DSM (v2.0)](https://github.com/CVPIA-OSC/fallRunDSM/releases/tag/v2.0)
* [Late Fall Run DSM (v1.0)](https://github.com/CVPIA-OSC/fallRunDSM/releases/tag/v1.0)
* [Winter Run DSM (v2.0)](https://github.com/CVPIA-OSC/winterRunDSM/releases/tag/v2.0)
* [Spring Run DSM (v2.0)](https://github.com/CVPIA-OSC/springRunDSM/releases/tag/v2.0)


R/data.R
#' @title Synthetic Years Index
#' @description We chose proxy years for the calibration time period of 1998-2017
#' by selecting the year from the 1980-1999 model inputs that most closely matched the
#' \href{https://cdec.water.ca.gov/reportapp/javareports?name=WSIHIST}{DWR water year index}
#' in both the San Joaquin and Sacramento Basins of the targeted calibration year.
#' @details
#' The 2019 calibration used a different selection of years using a similar methodology
#' @name year_indicies
NULL

#' @rdname year_indicies
#' @format NULL
"calibration_year_index"

#' @rdname year_indicies
#' @format NULL
"calibration_year_spawn_index"

#' @rdname year_indicies
#' @format NULL
"calibration_year_index_2019"

#' @rdname year_indicies
#' @format NULL
"calibration_year_spawn_index_2019"

#' @title Grandtab
#' @description GrandTab data is a compilation of sources estimating the late-fall,
#' winter, spring, and fall-run Chinook salmon populations for streams surveyed.
#' Estimates are provided by the California Department of Fish and Wildlife,
#' the US Fish and Wildlife Service, the California Department of Water Resources,
#' the East Bay Municipal Utilities District, the US Bureau of Reclamation,
#' the Lower Yuba River Management Team, and the Fisheries Foundation of California.
#' For more information see \href{https://wildlife.ca.gov/Conservation/Fishes/Chinook-Salmon/Anadromous-Assessment}{CDFW Anadromous Assessment}
#'
#' GrandTab data is not used for the Yuba River. Instead escapement data for the Yuba comes from
#' Vaki monitoring. See \href{https://cvpia-documents.s3.us-west-1.amazonaws.com/CVPIATemplate_AdultYubaRiver_Cordoleani.docx}{SIT proposal} for additional details.
#' @name grandtab
NULL

#' @rdname grandtab
#' @format NULL
#' @section Observed:
#' This data will be used in the fitness function to measure the difference
#' between model predictions and observed escapement. Missing values are NA and we
#' made all records less than 100 NA to account for a lack of confidence for counts
#' less than 100. For spring run, counts less than 100 are included. Feather and Yuba
#' Rivers do not separate spring and fall run, these estimate were scaled using
#' a mean proportion for the runs from 2010-2012 coded wire tag data.
"grandtab_observed"

#' @rdname grandtab
#' @format NULL
#' @section Imputed:
#'  This data will be used to calculate the number of juveniles during
#' the 20 year simulation. The GrandTab data is incomplete for many watersheds
#' during the 20 year period of calibration. For watersheds with no GrandTab data,
#' we used 40 as the default escapement value. For watersheds with incomplete data,
#' we used the mean escapement value.
"grandtab_imputed"

#' @title Mean Escapement 2013-2017
#' @description Mean Grandtab in-river escapement values for CVPIA SIT watersheds (2013-2017)
#' @details see \link{grandtab} for more information on Grandtab
"mean_escapement_2013_2017"


R/set_synth_years.R
#' Set Synthetic Year Series
#' @description Generates model params for running the life cycle models in calibration mode
#' @details This function modifies the baseline model params for a run using
#' the synthetic year series. More details about the synthetic year series can be found
#' here: \code{\link{calibration_year_index}}.
#'
#' Results of running \code{set_synth_years} for each run are cached and accessible via
#' \code{\link{fall_run_calibration}}, \code{\link{winter_run_calibration}},and
#' \code{\link{spring_run_calibration}}
#' @param params list object containing all arguments and coefficents for a life cycle model
#' @param spawn_years named array of synthetic year series for spawning inputs
#' @param years named array of synthetic year series for all other inputs
#' @examples
#' \dontrun{
#' params <- set_synth_years(fallRunDSM::params)
#' fallRunDSM::fall_run_model(mode = "calibrate", seeds = seeds, ..params = params)
#' }
#' @export
set_synth_years <- function(params,
                            spawn_years = DSMCalibrationData::calibration_year_spawn_index,
                            years = DSMCalibrationData::calibration_year_index) {

  params$freeport_flows <- params$freeport_flows[ , years]
  params$vernalis_flows <- params$vernalis_flows[ , years]
  params$stockton_flows <- params$stockton_flows[ , years]
  params$CVP_exports <- params$CVP_exports[ , years]
  params$SWP_exports <- params$SWP_exports[ , years]
  params$proportion_diverted <- params$proportion_diverted[ , , years]
  params$total_diverted <- params$total_diverted[ , , years]

  params$delta_proportion_diverted <- params$delta_proportion_diverted[ , years, ]
  params$delta_total_diverted <- params$delta_total_diverted[ , years, ]
  params$delta_inflow <- params$delta_inflow[ , years, ]
  params$proportion_flow_bypass <- params$proportion_flow_bypass[ , years, ]
  params$gates_overtopped <- params$gates_overtopped[ , years, ]
  params$avg_temp_delta <- params$avg_temp_delta[ , years, ]
  params$delta_habitat <- params$delta_habitat[ , years, ]


  params$prop_flow_natal <- params$prop_flow_natal[ , spawn_years]
  params$upper_sacramento_flows <- params$upper_sacramento_flows[ , years]
  params$vernalis_temps <- params$vernalis_temps[ , years]
  params$prisoners_point_temps <- params$prisoners_point_temps[ , years]
  params$degree_days <- params$degree_days[ , , spawn_years]
  params$avg_temp <- params$avg_temp[ , , years]
  params$spawning_habitat <- params$spawning_habitat[ , , spawn_years]
  params$inchannel_habitat_fry <- params$inchannel_habitat_fry[ , , years]
  params$inchannel_habitat_juvenile <- params$inchannel_habitat_juvenile[ , , years]
  params$floodplain_habitat <- params$floodplain_habitat[ , , years]
  params$weeks_flooded <- params$weeks_flooded[ , , years]
  params$sutter_habitat <- params$sutter_habitat[ , years]
  params$yolo_habitat <- params$yolo_habitat[ , years]

  return(params)
}



data-raw/synthetic_years.R
# remotes::install_github('flowwest/waterYearType')
library(waterYearType)
library(tidyverse)

target_sac <- water_year_indices %>%
  filter(location == 'Sacramento Valley', between(WY, 1997, 2017)) %>%
  select(WY, Index)

options_sac <- water_year_indices %>%
  filter(location == 'Sacramento Valley', between(WY, 1980, 2000)) %>%
  pull(Index)

names(options_sac) <- 1980:2000

target_sj <- water_year_indices %>%
  filter(location == 'San Joaquin Valley', between(WY, 1997, 2017)) %>%
  select(WY, Index)

options_sj <- water_year_indices %>%
  filter(location == 'San Joaquin Valley', between(WY, 1980, 2000)) %>%
  pull(Index)

names(options_sj) <- 1980:2000

synth_year_mapping <- map_chr(seq(target_sj$Index),
                              ~names(
                                which.min(
                                  abs(target_sj$Index[.] - options_sj) +
                                    abs(target_sac$Index[.] - options_sac))
                              )
)

calibration_proxy_year <- data.frame(year = 1997:2017, calibration_year = synth_year_mapping,
                                     sac_actual = target_sac$Index,
                                     sac_synth = options_sac[synth_year_mapping],
                                     sj_actual = target_sj$Index,
                                     sj_synth = options_sj[synth_year_mapping])

cor(calibration_proxy_year$sj_actual, calibration_proxy_year$sj_synth)
cor(calibration_proxy_year$sac_actual, calibration_proxy_year$sac_synth)

calibration_year_spawn_index <- setNames(calibration_proxy_year$calibration_year, 1997:2017)
calibration_year_index <- calibration_year_spawn_index[-1]

usethis::use_data(calibration_year_index, overwrite = TRUE)
usethis::use_data(calibration_year_spawn_index, overwrite = TRUE)

calibration_year_index_2019 <- c("1998" = "1998", "1999" = "1997", "2000" = "1993", "2001" = "1981", "2002" = "1989",
           "2003" = "1993", "2004" = "1993", "2005" = "1993", "2006" = "1998", "2007" = "1994",
           "2008" = "1988", "2009" = "1994", "2010" = "1985", "2011" = "1997", "2012" = "1985",
           "2013" = "1994", "2014" = "1992", "2015" = "1992", "2016" = "1989", "2017" = "1998")

calibration_year_spawn_index_2019 <- c("1997" = "1997", calibration_year_index_2019)

usethis::use_data(calibration_year_index_2019, overwrite = TRUE)
usethis::use_data(calibration_year_spawn_index_2019, overwrite = TRUE)


data-raw/explore-grandtab.R
library(fallRunDSM)
library(tidyverse)

watershed_order <- tibble(
  watershed = DSMscenario::watershed_labels,
  order = 1:31
)

# SacPas data
grandtab_raw <- read_csv("data-raw/Grandtab_Modified.csv")
unique(grandtab_raw$run)
# Explore Locations -----
unique(grandtab_raw$location)
# location "Butte Creek" for spring run uses carcus count when available then snorkel
butte <- c(
  "Butte Creek - Carcass" = "Butte Creek",
  "Butte Creek - Snorkel" = "Butte Creek",
  "Butte Creek" = "Butte Creek")

grandtab_raw %>%
  mutate(ll = butte[location]) %>%
  filter(!is.na(ll), startyear >= 1998) %>%
  select(startyear, location, count, run) %>%
  spread(startyear, count) %>%  View

# Battle Creek
# remove "Hatchery Transfers to Battle Creek - CNFH" = "Battle Creek" from battle creek sum
# remove " Battle Creek - CNFH "
battle <- c("Battle Creek" = "Battle Creek", "Battle Creek - CNFH" = "Battle Creek",
            "Battle Creek - Downstream of CNFH" = "Battle Creek",
            "Battle Creek - Upstream of CNFH" = "Battle Creek",
            "Hatchery Transfers to Battle Creek - CNFH" = "Battle Creek")

grandtab_raw %>%
  mutate(ll = battle[location]) %>%
  filter(!is.na(ll), startyear >= 1998) %>%
  select(startyear, location, count, run) %>%
  spread(startyear, count) %>%  View

grandtab_raw %>%
  filter(location == "Mokelumne River") %>% View

grandtab_raw %>%
  filter(location %in% names(battle), between(endyear, 2013, 2017)) %>%
  group_by(location, origin) %>%
  summarise(n())
summarise(mean=mean(count, na.rm=T))

# sacramento
# remove "Passing RBDD", "Downstream of RBDD", "Passing RBDD", "Upstream of RBDD"
grandtab_raw %>%
  filter(location %in% c("Mainstem - Downstream of RBDD",
                         "Mainstem - Upstream of RBDD",
                         "Mainstem",
                         "Passing RBDD",
                         "Downstream of RBDD",
                         "Upstream of RBDD"),
         startyear >= 1998, run == "Late-Fall") %>%
  select(startyear, minorbasin, location, count, run) %>%
  spread(startyear, count)


grandtab_raw %>%
  filter(location %in% c("Mainstem - Downstream of RBDD",
                         "Mainstem - Upstream of RBDD",
                         "Mainstem",
                         "Passing RBDD",
                         "Downstream of RBDD",
                         "Upstream of RBDD"),
         between(endyear, 2013, 2017)) %>%
  group_by(location, origin) %>%
  summarise(n())

grandtab_raw %>%
  filter(origin == "Redd Distribution")


data-raw/load_synth_years.R
remotes::install_github("CVPIA-OSC/DSMflow@main")
remotes::install_github("CVPIA-OSC/DSMtemperature@main")
remotes::install_github("CVPIA-OSC/DSMhabitat@main")
remotes::install_github("CVPIA-OSC/fallRunDSM@main")
# remotes::install_github("CVPIA-OSC/winterRunDSM@main")
# remotes::install_github("CVPIA-OSC/latefallRunDSM@main")
# remotes::install_github("CVPIA-OSC/springRunDSM@main")

# create the data to be cached ----
# fall run
fall_run_calibration_params <- set_synth_years(species = "fr")
usethis::use_data(fall_run_calibration_params, overwrite = TRUE)

# spring
# spring_run_calibration_params <- set_synth_years(species = "sr")
# usethis::use_data(spring_run_calibration_params, overwrite = TRUE)
#
# # winter
# winter_run_calibration_params <- set_synth_years(species = "wr")
# usethis::use_data(winter_run_calibration_params, overwrite = TRUE)
#
# late_fall_run_calibration_params <- set_synth_years(species = "lfr")
# usethis::use_data(late_fall_run_calibration_params, overwrite = TRUE)


data-raw/grandtab.R
library(fallRunDSM)
library(tidyverse)

watershed_order <- tibble(
  watershed = DSMscenario::watershed_labels,
  order = 1:31
)

# SacPas data
grandtab_raw <- read_csv("data-raw/Grandtab_Modified.csv")
unique(grandtab_raw$run)

# cleaned up location names
watershed_lookups <- c(
  "Mainstem - Downstream of RBDD" = "Upper Sacramento River",
  "Mainstem - Upstream of RBDD" = "Upper Sacramento River",
  "Mainstem" = "Upper Sacramento River",
  "Antelope Creek" = "Antelope Creek",
  "Battle Creek" = "Battle Creek",
  "Battle Creek - Downstream of CNFH" = "Battle Creek",
  "Battle Creek - Upstream of CNFH" = "Battle Creek",
  "Bear Creek" = "Bear Creek",
  "Big Chico" = "Big Chico Creek",
  "Butte Creek" = "Butte Creek",
  "Clear Creek" = "Clear Creek",
  "Cottonwood Creek" = "Cottonwood Creek",
  "Cow Creek" = "Cow Creek",
  "Deer Creek" = "Deer Creek",
  "Mill Creek" = "Mill Creek",
  "Paynes Creek" = "Paynes Creek",
  "Thomes Creek" = "Thomes Creek",
  "Bear River" = "Bear River",
  "Feather River" = "Feather River",
  "Yuba River" = "Yuba River",
  "American River" = "American River",
  "Calaveras River" = "Calaveras River",
  "Cosumnes River" = "Cosumnes River",
  "Mokelumne River" = "Mokelumne River",
  "Merced River" = "Merced River",
  "Stanislaus River" = "Stanislaus River",
  "Tuolumne River" = "Tuolumne River"
)

# Filter grandtab data ----
grandtab <- grandtab_raw %>%
  mutate(location2 = watershed_lookups[location]) %>%
  filter(!is.na(location2), endyear >= 1998, origin == "In-River") %>%
  group_by(run, watershed = location2, year = endyear) %>%
  summarise(count = sum(count, na.rm = TRUE)) %>%
  ungroup() |>
  filter(!(watershed == "Yuba River" & run %in% c("Fall", "Spring") &
             year %in% c(2004:2015, 2018, 2019, 2021))) |>
  mutate(method = "grandtab") |>
  glimpse()

# wrangle Yuba to match grandtab format ----
yuba_data <- read_csv("data-raw/yuba_escapement_values.csv") %>%
  rename(Spring = spring_run_escapement,
         Fall = fall_run_escapement) |>
  pivot_longer(cols = Spring:Fall, names_to = 'run', values_to = 'count') |>
  mutate(watershed = "Yuba River",
         method = "vaki") |>  glimpse()

grandtab_with_yuba_updates <- bind_rows(grandtab, yuba_data) |>
  filter(!is.na(year),  year < 2018) |> glimpse()

# Imputed Grandtab ----
# Fall
fall_spawn <- DSMhabitat::fr_spawn[['biop_2008_2009']][, 10, 1] != 0
fall_prop_feather_yuba <- 1 - mean(c(0.076777295, 0.056932196, 0.081441457))

grandtab_imputed_fall <- grandtab_with_yuba_updates %>%
  filter(run == "Fall") %>%
  select(-run) %>%
  right_join(watershed_order) %>%
  group_by(watershed, order) %>%
  mutate(
    mean = round(mean(count[count > 0], na.rm = TRUE)),
    count2 = case_when(
      !fall_spawn[watershed] ~ 0,
      fall_spawn[watershed] & is.nan(mean) ~ 40,
      fall_spawn[watershed] & count == 0 ~ mean,
      fall_spawn[watershed] & is.na(count) ~ mean,
      TRUE ~ count
    )
  ) %>%
  ungroup() %>%
  select(watershed, count = count2, year, order, method) %>%
  mutate(count = case_when(watershed == "Feather River" ~ round(count * fall_prop_feather_yuba),
                           watershed == "Yuba River" & method == "grandtab" ~ round(count * fall_prop_feather_yuba),
                           T ~ count)) %>%
  select(-method) %>%
  spread(year, count) %>%
  arrange(order) %>%
  select(-watershed, -order, -`<NA>`) %>%
  as.matrix()

rownames(grandtab_imputed_fall) <- watershed_order$watershed

# Winter
winter_spawn <- DSMhabitat::wr_spawn[['biop_2008_2009']][ ,10, 1] != 0
upsac_wr <- grandtab %>%
  filter(run == "Winter", watershed == "Upper Sacramento River") %>%
  select(-run)

bat_wr <- upsac_wr %>%
  mutate(watershed = "Battle Creek",
         count = round((600/mean(count))* count))

grandtab_imputed_winter <- bind_rows(upsac_wr, bat_wr) %>%
  select(-method) %>%
  right_join(watershed_order) %>%
  spread(year, count) %>%
  select(-`<NA>`) %>%
  arrange(order) %>%
  select(-watershed, -order) %>%
  as.matrix()

grandtab_imputed_winter[which(is.na(grandtab_imputed_winter))] <- 0

rownames(grandtab_imputed_winter) <- watershed_order$watershed

# Spring
# TODO what to do about tiny mean values?
spring_spawn <- DSMhabitat::sr_spawn[['biop_2008_2009']][,10,1] != 0
spring_prop_feather_yuba <- mean(c(0.076777295, 0.056932196, 0.081441457))

grandtab_imputed_spring <- grandtab_with_yuba_updates %>%
  filter(run == "Spring") %>%
  select(-run) %>%
  right_join(watershed_order) %>%
  group_by(watershed, order) %>%
  mutate(
    mean = round(mean(count[count > 0], na.rm = TRUE)),
    count2 = case_when(
      !spring_spawn[watershed] ~ 0,
      spring_spawn[watershed] & is.nan(mean) ~ 40,
      spring_spawn[watershed] & count == 0 ~ mean,
      spring_spawn[watershed] & is.na(count) ~ mean,
      TRUE ~ count
    )
  ) %>%
  ungroup() %>%
  select(watershed, count = count2, year, order) %>%
  spread(year, count) %>%
  arrange(order) %>%
  select(-watershed, -order, -`<NA>`) %>%
  as.matrix()

rownames(grandtab_imputed_spring) <- watershed_order$watershed

grandtab_imputed_spring["Feather River", ] <- round(grandtab_imputed_fall["Feather River", ]/ fall_prop_feather_yuba * spring_prop_feather_yuba)
grandtab_imputed_spring["Yuba River", 1:6] <- round(grandtab_imputed_fall["Yuba River",  1:6]/ fall_prop_feather_yuba * spring_prop_feather_yuba) # use grantab with prop spring scaling for years that there is no vaki
grandtab_imputed_spring["Yuba River", 19:20] <- round(grandtab_imputed_fall["Yuba River", 19:20]/ fall_prop_feather_yuba * spring_prop_feather_yuba) # use grantab with prop spring scaling for years that there is no vaki

# Late-Fall
late_fall_spawn <- DSMhabitat::lfr_spawn[['biop_2008_2009']][,10,1] != 0

grandtab_imputed_late_fall <- grandtab %>%
  filter(run == "Late-Fall") %>%
  select(-run) %>%
  right_join(watershed_order) %>%
  group_by(watershed, order) %>%
  mutate(
    mean = round(mean(count[count > 0], na.rm = TRUE)),
    count2 = case_when(
      !late_fall_spawn[watershed] ~ 0,
      late_fall_spawn[watershed] & is.nan(mean) ~ 40,
      late_fall_spawn[watershed] & count == 0 ~ mean,
      late_fall_spawn[watershed] & is.na(count) ~ mean,
      TRUE ~ count
    )
  ) %>%
  ungroup() %>%
  select(watershed, count = count2, year, order) %>%
  spread(year, count) %>%
  arrange(order) %>%
  select(-watershed, -order, -`<NA>`) %>%
  as.matrix()

rownames(grandtab_imputed_late_fall) <- watershed_order$watershed

grandtab_imputed <- list(fall = grandtab_imputed_fall,
                         late_fall = grandtab_imputed_late_fall,
                         winter = grandtab_imputed_winter,
                         spring = grandtab_imputed_spring)

usethis::use_data(grandtab_imputed, overwrite = TRUE)

# Grandtab Observed -----

# Fall
grandtab_observed_fall <- grandtab_with_yuba_updates %>%
  filter(run == "Fall") %>%
  select(-run) %>%
  filter(count > 100) %>%
  right_join(watershed_order) %>%
  mutate(count = case_when(watershed == "Feather River" ~ round(count * fall_prop_feather_yuba),
                           watershed == "Yuba River" & method == "grandtab" ~ round(count * fall_prop_feather_yuba),
                           T ~ count)) %>%
  select(-method) %>%
  spread(year, count) %>%
  select(-`<NA>`) %>%
  arrange(order) %>%
  select(-watershed, -order) %>%
  as.matrix()

rownames(grandtab_observed_fall) <- watershed_order$watershed

# Winter
grandtab_observed_winter <- grandtab_imputed_winter

# Spring
grandtab_observed_spring <- grandtab %>%
  filter(run == "Spring") %>%
  select(-run, -method) %>%
  right_join(watershed_order) %>%
  spread(year, count) %>%
  select(-`<NA>`) %>%
  arrange(order) %>%
  select(-watershed, -order) %>%
  as.matrix()

rownames(grandtab_observed_spring) <- watershed_order$watershed

grandtab_observed_spring["Feather River", ] <- round(grandtab_observed_fall["Feather River", ]/ fall_prop_feather_yuba * spring_prop_feather_yuba)
grandtab_observed_spring["Yuba River", ] <- round(grandtab_observed_fall["Yuba River", ]/ fall_prop_feather_yuba * spring_prop_feather_yuba)
grandtab_observed_spring["Yuba River", 1:6] <- round(grandtab_observed_fall["Yuba River",  1:6]/ fall_prop_feather_yuba * spring_prop_feather_yuba) # use grantab with prop spring scaling for years that there is no vaki
grandtab_observed_spring["Yuba River", 19:20] <- round(grandtab_observed_fall["Yuba River", 19:20]/ fall_prop_feather_yuba * spring_prop_feather_yuba) # use grantab with prop spring scaling for years that there is no vaki


# Late-Fall
grandtab_observed_late_fall <- grandtab %>%
  filter(run == "Late-Fall") %>%
  select(-run, -method) %>%
  filter(count > 100) %>%
  right_join(watershed_order) %>%
  spread(year, count) %>%
  select(-`<NA>`) %>%
  arrange(order) %>%
  select(-watershed, -order) %>%
  as.matrix()

rownames(grandtab_observed_late_fall) <- watershed_order$watershed

grandtab_observed <- list(fall = grandtab_observed_fall,
                          late_fall = grandtab_observed_late_fall,
                          winter = grandtab_observed_winter,
                          spring = grandtab_observed_spring)
usethis::use_data(grandtab_observed, overwrite = TRUE)

# check species presence ---
# should_be_missing <- DSMscenario::watershed_labels[
#   !as.logical(DSMhabitat::watershed_species_present[1:31, ]$fr *
#                 DSMhabitat::watershed_species_present[1:31,]$spawn)]
# missing <- DSMscenario::watershed_labels[is.nan(rowMeans(grandtab_observed_fall, na.rm = T))]
# all(should_be_missing %in% missing)
# missing <- DSMscenario::watershed_labels[is.nan(rowMeans(grandtab_imputed_fall, na.rm = T))]
# all(should_be_missing %in% missing)
#
# should_be_missing <- DSMscenario::watershed_labels[
#   !as.logical(DSMhabitat::watershed_species_present[1:31, ]$lfr *
#                 DSMhabitat::watershed_species_present[1:31,]$spawn)]
# missing <- DSMscenario::watershed_labels[is.nan(rowMeans(grandtab_observed_late_fall, na.rm = T))]
# all(should_be_missing %in% missing)
# missing <- DSMscenario::watershed_labels[is.nan(rowMeans(grandtab_imputed_late_fall, na.rm = T))]
# all(should_be_missing %in% missing)
#
# should_be_missing <- DSMscenario::watershed_labels[
#   !as.logical(DSMhabitat::watershed_species_present[1:31, ]$sr *
#                 DSMhabitat::watershed_species_present[1:31,]$spawn)]
# missing <- DSMscenario::watershed_labels[is.nan(rowMeans(grandtab_observed_spring, na.rm = T))]
# all(should_be_missing %in% missing)
# missing <- DSMscenario::watershed_labels[is.nan(rowMeans(grandtab_imputed_spring, na.rm = T))]
# all(should_be_missing %in% missing)
#
# should_be_missing <- DSMscenario::watershed_labels[
#   !as.logical(DSMhabitat::watershed_species_present[1:31, ]$wr *
#                 DSMhabitat::watershed_species_present[1:31,]$spawn)]
# missing <- DSMscenario::watershed_labels[is.nan(rowMeans(grandtab_observed_winter, na.rm = T))]
# all(should_be_missing %in% missing)
# missing <- DSMscenario::watershed_labels[is.nan(rowMeans(grandtab_imputed_winter, na.rm = T))]
# all(should_be_missing %in% missing)

# Adult Seeds -----
# Jim used "Battle Creek - Downstream of CNFH" for "Battle Creek"
mean_escapement_2013_2017 <- grandtab_with_yuba_updates %>%
  filter(between(year, 2013, 2017)) %>%
  group_by(watershed, run) %>%
  summarise(mean = round(mean(count, na.rm = TRUE))) %>%
  spread(run, mean) %>%
  right_join(watershed_order) %>%
  arrange(order) %>%
  select(-order) %>%
  ungroup()

usethis::use_data(mean_escapement_2013_2017, overwrite = TRUE)
